### ISSUE TITLE
Security Bug(Group-11) : Using database as auth file gives access to those who don't have new auth file 

### ISSUE THREAD

**Issue by Pratham-Sharda** [IITGN-CS431/project-cns#242]
@bichhawat Sir as the current code is just using database as its auth file ,

There might be a case where the bank closed and then is again run but as the auth file (newly created) will be same as before (assuming there were no changes in the database) so ideally it should be that only those atm who have these newly created auth file should be able to connect to bank , but as there was no change in database all those who had the earlier database will still be able to connect to the bank run now (Thing to note is that these atms were not provided with the newly created auth file but still are able to connect to bank)

Shouldn't this be considered as a security issue as the old auth file is giving access to the bank even when they don't have the newly generated auth file? 

The purpose of auth file creation is defeated as old ATMs(without the access to new auth file) are able to connect to the bank.

**Comment by bichhawat** [comment]
This seems like a legitimate issue. Can you please provide some example showing how this works and what is expected?

**Comment by Pratham-Sharda** [comment]
![image](https://github.com/user-attachments/assets/42cd926b-561d-4e07-969d-41801c4a3c87)
Sir so as you can see i am running a atm in folder build2 where the auth file(auth_backup) being used is created at 13:46[Which signifies that this auth file was created during earlier run of bank]
![image](https://github.com/user-attachments/assets/e508f9ef-7e6d-48eb-b0a5-d889875861ee)

Now as you can see in the new bank run which replaced the old auth_backup and created a new one at 13:53 as shown in image below ,This bank created new auth file which was not given in folder build2 but still the ATM in build2 folder is able to connect to the current run of bank using its old auth_backup[showing that ideally without the new auth file the atm in build2 should not be able to connect to bank run at 13:53 but still can access the bank using old auth file created at 13:46]

new auth_backup created at 13:53 
![image](https://github.com/user-attachments/assets/8303ee74-73be-4249-a2cf-8ec1a8311871)

**Comment by PrateekG93** [comment]
The authentication in our system is entirely managed through SSL/TLS certificates, not the auth file. Here’s how it works:

**SSL/TLS Server Authentication:**

The bank server uses an SSL/TLS certificate verified by a trusted Certificate Authority (CA) to prove its identity to clients (ATMs) during the SSL handshake.

**Client (ATM) Authentication via Mutual TLS (mTLS):**

We use Mutual TLS, meaning each ATM must also present a valid certificate. Only ATMs with trusted certificates can connect. If a certificate is invalid or revoked, the connection is denied.

**Role of the auth File:**

The auth file is not involved in authentication; it’s solely for storing session data. Therefore, changes to the auth file do not impact which ATMs can connect.

In short, only ATMs with valid SSL/TLS certificates are allowed to connect, regardless of the auth file’s state. The concern about “old ATMs with an outdated auth file” does not apply, as certificate-based authentication ensures that only authorized ATMs can access the bank server.

**Comment by bichhawat** [comment]
@Pratham-Sharda Please take a look at the comment and discuss if you have any questions

**Comment by Pratham-Sharda** [comment]
Sir here the auth file was required to authenticate the ATM and as mentioned in spec auth file authenticate the ATM and it wasn't mentioned on the read me page earlier about what the use of auth file creation is in the project. (And thus the auth file usage for authenticating ATM is required according to spec)

**Comment by bichhawat** [comment]
Got it. @PrateekG93 kindly mention this feature in the README to clarify.

**Comment by PrateekG93** [comment]
Okay sir, I will
